client/galaxy/scripts/components/Workflow/Editor/Node.vue

```diff
@@ -69,9 +91,20 @@ export default {
             }
             return null;
         },
+        popoverId() {
+            return `popover-${this.id}-${this._uid}`;
```

Comment:
Can you explain why both ids are needed?

Reply:
On workflow editor, a toolbox can appear multiple times. If we use the same popover id for multiple (and same) tools, then it is shown only for one box and other tools with same ids do not have popover. Therefore to make ids unique for each toolbox, I am appending `this._uid` id

Reply:
The `id` should be unique. I wonder if it has been set properly.

Reply:
The `id` is unique but its the tool's id. I looked at it further and found out that in method `mountWorkflowNode`  (https://github.com/galaxyproject/galaxy/blob/577443181c9d814d14280b5e3c30259ba2756860/client/galaxy/scripts/mvc/workflow/workflow-manager.js#L140) 
the `id` is actually the `content_id` (tool's id) and not the id of the parent node (which is unique and takes form as `wf-node-step ...`). Once this call is over, only then these ids are assigned in `add_node` method (https://github.com/galaxyproject/galaxy/blob/577443181c9d814d14280b5e3c30259ba2756860/client/galaxy/scripts/mvc/workflow/workflow-manager.js#L114)

Therefore the unique ids of each node is not available in `Node.vue` . To fix that I created one counter in `workflow-manager.js` where all the nodes are created and `Node.vue` is invoked. I use this counter in `Node.vue` to assign unique ids to each popover.